api/v2/views/machine_request.py

```diff
@@ -1,67 +1,110 @@
     filter_fields = ('status__id', 'status__name', 'new_machine_owner__username')
+    ordering_fields = ('start_date', 'end_date', 'new_machine_owner__username')
+    ordering = ('-start_date',)
 
     def perform_create(self, serializer):
+
+        q = MachineRequest.objects.filter(
+            (
+                Q(created_by__id=self.request.user.id) &
+                Q(instance_id=serializer.validated_data['instance'].id) &
+                ~Q(status__name="failed") &
+                ~Q(status__name="rejected") &
+                ~Q(status__name="completed") &
+                ~Q(status__name="closed")
+            ))
+
+        if len(q) > 0:
+            message = "Only one active request is allowed per provider."
+            raise rest_exceptions.MethodNotAllowed('create', detail=message)
+
         # NOTE: An identity could possible have multiple memberships
         # It may be better to directly take membership rather than an identity
         identity_id = serializer.initial_data.get("identity")
-        new_provider_id= serializer.initial_data['new_machine_provider']
         new_owner=self.request.user
         parent_machine = serializer.validated_data['instance'].provider_machine
+
+        # TODO: This is a hack that can be removed POST-ll (When MachineRequest validates new_machine_provider)
+        new_provider = parent_machine.provider  # <--HACK!
+
         status, _ = StatusType.objects.get_or_create(name="pending")
+        new_machine_provider = Provider.objects.filter(id=new_provider.id)
+        new_machine_owner = AtmosphereUser.objects.filter(id=new_owner.id)
+        parent_machine = ProviderMachine.objects.filter(id=parent_machine.id)
+
+        if new_machine_provider.count():
+            new_machine_provider = new_machine_provider[0]
+        else:
+            raise rest_exceptions.ParseError(detail="Could not retrieve new machine provider.")
+
+        if new_machine_owner.count():
+            new_machine_owner = new_machine_owner[0]
+        else:
+            raise rest_exceptions.ParseError(detail="Could not retrieve new machine owner.")
+
+        if parent_machine.count():
+            parent_machine = parent_machine[0]
+        else:
+            raise rest_exceptions.ParseError(detail="Could not retrieve parent machine.")
+
         try:
             membership = IdentityMembership.objects.get(identity=identity_id)
             instance = serializer.save(
                 membership=membership,
                 status=status,
-                old_status="pending",
                 created_by=self.request.user,
-                #new_machine_provider_id = new_provider_id,
-                new_machine_owner = new_owner,
-                #parent_machine_id=parent_machine_id
+                new_machine_provider=new_provider,
+                new_machine_owner=new_owner,
+                old_status="pending",  # TODO: Is this required or will it default to pending?
```

Comment:
To chime in on this comment...

I believe in testing that `old_status` will _default_ to `"pending"`. The important thing is that it is the _default_ or `"pending"` - but **not** `"processing"` or we get `"processing - processing"` (as noted/discussion [ATMO-1104](https://pods.iplantcollaborative.org/jira/browse/ATMO-1104)). 

Also discussed in the related PR #86.